### PR TITLE
Fix broken contribution links in README

### LINKED ISSUE #3907: The "submit a blog" link is dead
In the main [README.md](https://github.com/open-telemetry/opentelemetry.io/blob/main/README.md), the link to "Submit a new blog post" is broken: https://github.com/open-telemetry/opentelemetry.io/blob/main/README.md?plain=1#L21
It points to https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/contributing/blog (no extension), but I think it should point to https://github.com/open-telemetry/opentelemetry.io/blob/main/content/en/docs/contributing/blog.md (added `.md`) or directly to https://opentelemetry.io/docs/contributing/blog/. 

Happy to send a PR, just not sure which one is preferred, the opentelemetry.io link or the link to the markdown file.

### PR DESCRIPTION
**Pull request by theletterf**
- Fixes #3907

Replacing relative links (that won't render in GitHub) with absolute URLs. Open to suggestions for a better approach (I'd rather not send folks to the docs source if we've a rendered site).

### CODE REVIEW

**Review by chalin — APPROVED** [review]
Yep, this is the way to go. Thanks for the fixes ✨

**Comment by chalin** [comment]
Btw, if you run the following and no errors are reported, then we're good:

```console
$ npm run _check:links--md
```

**Comment by theletterf** [comment]
Thanks! Wouldn't `text-and-fix` from the CLI also perform that check?

**Comment by chalin** [comment]
No, back when I added that check I chose to leave it out of the usual CI pipeline & commands.

My preference would be to publish, e.g. under `/site`, all of the repo `*.md` files so that we can use htmltest + external link caching instead of the markdown link checker (which was a stop-gap measure). So much fun functionality to add!

File: README.md [hunk 1]
```diff
@@ -7,8 +7,8 @@ is hosted on [Netlify][].
 ## Get involved
 
 To learn how to contribute fixes and new content to this project, read the
-[Contributor's guide](/content/en/docs/contributing), which includes a style
-guide and useful information on the review process.
+[Contributor's guide](https://opentelemetry.io/docs/contributing/), which
```
Comment (svrnm) [inline README.md:10]: make sure you use relative links

File: README.md [hunk 2]
```diff
@@ -18,7 +18,7 @@ missing [let us know][].
 ### Submit a blog post
 
 For guidance on how to write and submit a blog post, see
-[Submit a blog post](/content/en/docs/contributing/blog).
+[Submit a blog post](https://opentelemetry.io/docs/contributing/blog/).
```
Comment (svrnm) [inline README.md:21]: make sure you use relative links

File: README.md (hunk 1 above)
Comment (svrnm) [reply README.md:10]:
```suggestion
[Contributor's guide](/docs/contributing/), which
```

File: README.md (hunk 2 above)
Comment (svrnm) [reply README.md:21]:
```suggestion
[Submit a blog post](/docs/contributing/blog/).
```

**Review by svrnm — APPROVED** [review]

**Pull request merged**